rs/consensus/src/consensus/validator.rs

```diff
@@ -956,14 +1034,11 @@ impl Validator {
                 // proposals proceed to be checked normally.
             }
 
-            // Skip validation and drop the block if it has a higher rank than a
-            // known valid block. Note that this must happen after we first allow
-            // "block with notarization" validation (see above). Otherwise we may
-            // get stuck when a block maker equivocates.
-            if let Some(Some(min_rank)) = known_qualified_ranks.get(&proposal.height()) {
-                if proposal.rank() > *min_rank {
-                    // Skip them instead of removal because we don't want to end up
-                    // requesting these artifacts again.
+            // Skip validation if proposal has a higher rank than a known
+            // valid block. We skip the block instead of removing it because
+            // a higher-rank proposal might still be notarized in the future.
+            if let Some(min_rank) = valid_ranks.get_lowest_rank(proposal.height()) {
```

Comment:
If we have a validated block that has both an equivocation proof and a notarization, we will continue to validate blocks with a higher rank. I don't think it's an issue because we wouldn't try to notarize them, but maybe we could include that exception in the comment

Reply:
I added a note in the comment. I also changed the comment on `valid_ranks` to make clear it's disjoint with `disqualified_ranks`.